app/widget/containers/main_body/content/settings/settings_tabs/team_member_tab/team_member_table_group.py

```diff
@@ -0,0 +1,104 @@
+from PySide2.QtCore import QAbstractTableModel, Qt
+from PySide2.QtWidgets import (
+    QAbstractItemView,
+    QGroupBox,
+    QHeaderView,
+    QTableView,
+    QVBoxLayout,
+    QWidget,
+)
+from schemas.db_schema import TeamMemberDbSchema
+
+from app.db.queries import get_team_members, upsert_team_members
+
+
+class TeamMemberTable(QAbstractTableModel):
+    def __init__(self):
+        super().__init__()
+        self.horizontal_headers = ["id", "imię i nazwisko", "specjalność"]
+        self._data = [[tm.id, tm.name, tm.function] for tm in get_team_members()]
+
+    def data(self, index, role):
+        if role == Qt.DisplayRole:
+            return self._data[index.row()][index.column()]
+
+    def setData(self, index, value, *args):
+        self._data[index.row()][index.column()] = value
+        self.dataChanged.emit(index, index, [Qt.DisplayRole])
+        return True
+
+    def is_last_row_valid(self) -> bool:
+        last_team_member = self._data[-1]
+        return last_team_member[1] and last_team_member[2]
+
+    def append_new_team_member(self):
+        last_team_member = self._data[-1]
+        if self.is_last_row_valid():
+            new_team_member = [last_team_member[0] + 1, "", ""]
```

Comment:
It would be good to know the structure of this `_data` -> from line 19 I see that the structure is:

```
id, name, function
```
and we check the defined schema below in the function `save_team_member_list`. Appending new team member could be also done here like:

```
TeamMemberDbSchema(last_team_member[0] +1, name=“", function=“”
```
right? Though I am still not sure about this `+1` ;-)

Reply:
1. Probably I'd be convinient to have it as pydantic model, instead of just list. But this widget is design to work with `int` as given cell index (row, column). Which means that we need to address given cell using `int` not ie. column name. 
2. Another idea is to use `namedtuple` which supports both indexing and calling by names. But as tuple it's immutable, and this feature doesn't allow us to update in when user change value.
3. We save team member data when on event, and this gather all introduced changes, to save it into db. `+1`. Creates new row index, which works as PK in db. 

But indeed all this logic is not very clear. Let me refactor it a bit.

Reply:
But, after second thought, I decided to rewrite this part completely. It's better to use library model which covers all our needs to use custom one.